Test/COD50303.TXT

```diff
@@ -0,0 +1,424 @@
+
+    [Test]
+    PROCEDURE TestWritingDifferentDataTypes@1();
+    VAR
+      ExpectedChar@1002 : Char;
+      ExpectedBigInt@1003 : BigInteger;
+      ExpectedDec@1004 : Decimal;
+    BEGIN
+      LibraryLowerPermissions.SetO365Basic;
+      DotNet_Encoding.UTF32;
+      CreateStream;
+      DotNet_BinaryWriter.BinaryWriterWithEncoding(DotNet_Stream, DotNet_Encoding);
+      DotNet_BinaryWriter.WriteChar('A');
+      DotNet_BinaryWriter.WriteChar('B');
+      DotNet_BinaryWriter.WriteByte(1);
+      DotNet_BinaryWriter.WriteInt16(2);
+      DotNet_BinaryWriter.WriteInt32(3);
+      DotNet_BinaryWriter.WriteUInt16(4);
+      DotNet_BinaryWriter.WriteUInt32(5);
+      DotNet_BinaryWriter.WriteBoolean(TRUE);
+      DotNet_BinaryWriter.WriteDecimal(6);
+      DotNet_BinaryWriter.WriteString('CDE');
+      CreateExampleByteArray;
+      DotNet_BinaryWriter.WriteArray(DotNet_ByteArray);
+      CreateExampleCharArray;
+      DotNet_BinaryWriter.WriteArray(DotNet_CharArray);
+      DotNet_BinaryWriter.Flush();
+      ExpectedBigInt := 71;
+      Assert.AreEqual(ExpectedBigInt, DotNet_Stream.Length, 'Stream length check failed');
+      DotNet_Stream.Seek(0, 0);
+      DotNet_BinaryReader.BinaryReaderWithEncoding(DotNet_Stream, DotNet_Encoding);
+      ExpectedChar := 'A';
+      Assert.AreEqual(ExpectedChar, DotNet_BinaryReader.ReadChar, 'ReadChar check failed');
+      ExpectedChar := 'B';
+      Assert.AreEqual(ExpectedChar, DotNet_BinaryReader.ReadChar, 'ReadChar check failed');
+      Assert.AreEqual(1, DotNet_BinaryReader.ReadByte, 'ReadByte check failed');
+      Assert.AreEqual(2, DotNet_BinaryReader.ReadInt16, 'ReadInt16 check failed');
+      Assert.AreEqual(3, DotNet_BinaryReader.ReadInt32, 'ReadInt32 check failed');
+      Assert.AreEqual(4, DotNet_BinaryReader.ReadUInt16, 'ReadUInt16 check failed');
+      Assert.AreEqual(5, DotNet_BinaryReader.ReadUInt32, 'ReadUInt32 check failed');
+      Assert.AreEqual(TRUE, DotNet_BinaryReader.ReadBoolean, 'ReadBoolean check failed');
+      ExpectedDec := 6;
+      Assert.AreEqual(ExpectedDec, DotNet_BinaryReader.ReadDecimal, 'ReadDecimal check failed');
+      Assert.AreEqual('CDE', DotNet_BinaryReader.ReadString, 'ReadString check failed');
+      CLEAR(DotNet_ByteArray);
+      DotNet_BinaryReader.ReadBytes(4, DotNet_ByteArray);
+      CheckArrayItems(DotNet_ByteArray, DotNet_ExpectedByteArray);
+      CLEAR(DotNet_CharArray);
+      DotNet_BinaryReader.ReadChars(4, DotNet_CharArray);
+      CheckArrayItems(DotNet_CharArray, DotNet_ExpectedCharArray);
+
+      DotNet_BinaryReader.Close;
+      DotNet_BinaryWriter.Close;
+      DotNet_BinaryReader.Dispose;
+      DotNet_BinaryWriter.Dispose;
+    END;
+
+    [Test]
+    PROCEDURE TestWriteBool@2();
+    BEGIN
+      //Based on https://github.com/dotnet/corefx/ BinaryWriter.WriteTests.cs BinaryWriter_WriteBoolTest
+      DotNet_MemoryStream.InitMemoryStream;
+      DotNet_MemoryStream.CastToStream(DotNet_Stream);
```

Comment:
The code is not compiling.
`CastToStream` is not a method/procedure in Codeunit 3009, the type of `DotNet_MemoryStream` :)

Please, provide the missing method.

Reply:
Just saw that the required changes are in another PR.

I will look into it now :)

Reply:
Hello,

Yes, it seems I forgot, that I used DotNet_MemoryStream modifications from another PR. Originally I hoped, that first PR will be approved before processing this one, well I was wrong :). Since CastToStream is used only for tests, I think I can modify test code to avoid using this function.

Reply:
No, it's not necessary 🙂

I'll have a look at your other PR and then proceed with this one.

In such cases, if possible, note such dependencies so that the PRs are processed in the right order.